job-profile-wrangler/src/main/java/org/folio/graph/GraphReader.java

```diff
@@ -0,0 +1,219 @@
+package org.folio.graph;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.RepoObject;
+import org.folio.graph.edges.MatchRelationshipEdge;
+import org.folio.graph.edges.NonMatchRelationshipEdge;
+import org.folio.graph.edges.RegularEdge;
+import org.folio.graph.nodes.ActionProfileNode;
+import org.folio.graph.nodes.JobProfileNode;
+import org.folio.graph.nodes.MappingProfileNode;
+import org.folio.graph.nodes.MatchProfileNode;
+import org.folio.graph.nodes.Profile;
+import org.jgrapht.Graph;
+import org.jgrapht.graph.SimpleDirectedGraph;
+import org.jgrapht.nio.Attribute;
+import org.jgrapht.nio.dot.DOTImporter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.folio.graph.GraphWriter.DOT_FILE_PATTERN;
+
+/**
+ * The GraphReader class provides methods to read and search graphs from DOT files.
+ * It uses the JGraphT library to represent and manipulate the graphs.
+ */
+public class GraphReader {
+  private static final Logger LOGGER = LogManager.getLogger();
+  private static final DOTImporter<Profile, RegularEdge> DOT_IMPORTER = new DOTImporter<>();
+
+  static {
+    // Configure the DOTImporter to create vertices based on the "name" attribute
+    DOT_IMPORTER.setVertexWithAttributesFactory((id, attrs) -> {
+      Map<String, String> normalAttrs = normalizeAttributes(attrs);
+      String name = normalAttrs.get("name");
+      switch (name) {
+        case "Job Profile" -> {
+          return JobProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Match Profile" -> {
+          return MatchProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Action Profile" -> {
+          return ActionProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Mapping Profile" -> {
+          return MappingProfileNode.fromAttributes(id, normalAttrs);
+        }
+        default -> {
+          LOGGER.error("Unrecognized profile: '{}'", normalAttrs);
```

Comment:
@okolawole-ebsco  Do we need here "error" level, or "warn" one will be enough? And one more point - maybe let's add here some more information ("id" etc)?

Reply:
I changed it to warning